src/js/control-bar/progress-control/seek-bar.js

```diff
@@ -116,8 +117,12 @@ class SeekBar extends Slider {
     }
 
     if (liveTracker && liveTracker.seekableEnd() === Infinity) {
-      this.disable();
-    } else {
+      if (this.player_.readyState() === 0) {
```

Comment:
How is the readyState === 0 if we already know the seekableEnd?

Reply:
I suppose the confusing part is that we would want the seekBar to be disabled only when we know we're live right? Rather than when we don't know?

Reply:
Basically, my change is for us to only re-enable the seek bar if it was disabled due to the seekable end being infinity.

Reply:
right, but what does the readyState check add to that? I believe seekable end can also be reported as Infinity on Android (which this case used to also cover)

Reply:
readyState of 0 means that no data has loaded, so, any value for seekable end is meaningless but the way that method is implemented, it'll still return Infinity. So, without the changes, it means that we would disable the progress control due to seekable end being infinity and then subsequently re-enable it when the video metadata loads. This means that if you disable the progress control yourself, it'll get re-enabled from under you. (We should also potentially check to see if we're currently disabled not via this codepath and then do nothing)

Reply:
To be more clear, I think `disabledForLive_` is valuable, as an indication that we disabled the seekBar and we should only re-enable again if it is `true`.

However, in the case where metadata takes a while to load we would actually want to disable the seek bar until we know if we are live, assuming we use the flag to check that is why we disabled it. But we _also_ want to disable the seek bar if we're on Android where DVR is not reliable because seekable end can be reported as Infinity.

I'm not sure exactly what the best course of action would be for the scenario where the liveTracker is disabled by a user and we're on Android. Should the progress bar show without allowing seeking?

Reply:
I think it's fine if the control bar isn't disabled in this case until the metadata is loaded, especially since it matches the current behavior. I think being permissive is a better side to err on in this case.

Reply:
> I think it's fine if the control bar isn't disabled in this case until the metadata is loaded

the Android case?

Reply:
That might work out, except for the initial setup, which we could put in an explicit case for.

Throwing out ideas: could we just explicitly have a flag on the liveTracker which determines if the seekbar should be enabled and use that in the constructor for seekBar? That would still allow `seekBar.disable` I believe

Reply:
Wouldn't that be the same as what we have now? Whether the check is here or in liveTracker, it'll still disable/enable incorrectly.
Also, I don't think that the livetracker should apply to the seekbar until/if we're in a livestream. So, I guess we could also check the duration and if it's not Infinity/-1 ignore it?

Reply:
https://github.com/videojs/video.js/pull/5721#issuecomment-451307233